### PR TITLE
[DOC] Add block quotes

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Also adjusts which blocks may be nested.

### CODE REVIEW

File: doc/rdoc/markup_reference.rb [hunk 1]
```diff
@@ -169,6 +173,49 @@
...
 #
+# ==== Block Quotes
+#
+# You can use the characters <tt>>>></tt> (unindented),
+# followed by indented text, to treat the text
+# as a {block quote}[https://en.wikipedia.org/wiki/Block_quotation]:
+#
+# Example input:
+#
+#   >>>
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:184]: The `>>>` here doesn't seem to correctly generate for me. I'm not sure what the best solution to this is.

Screen Shot 2022-07-28 at 1 31 33 PM

Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb:184]: I got a period, too.  I tried escaping one or all three characters; didn't work.  May be a bug.

Comment (nobu) [reply doc/rdoc/markup_reference.rb:184]: I think fixed it now.
Can you rebase this PR?

Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb:184]: Thanks, @nobu, for the fix.  I don't know rebase.  Maybe @peterzhu2118 can help?

Comment (peterzhu2118) [reply doc/rdoc/markup_reference.rb:184]: I did a rebase. The bug is fixed. Thanks @nobu!

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**